File: module thanh toán + giao hàng/app/controllers/PaymentController.php
<?php

require_once __DIR__ . '/../services/PaymentService.php';

class PaymentController
{
    private $paymentService;

    public function __construct()
    {
        $this->paymentService = new PaymentService();
    }

    public function checkout()
    {
        include __DIR__ . '/../views/payment/checkout.php';
    }

    public function processCheckout()
    {
        $orderId = $_POST['order_id'] ?? null;
        $amount = $_POST['amount'] ?? null;

        if (!$orderId || !$amount) {
            echo "Thiếu dữ liệu thanh toán";
            return;
        }

        $this->paymentService->processCOD($orderId, $amount);
        include __DIR__ . '/../views/payment/confirm.php';
    }

    public function confirm()
    {
        $paymentId = $_POST['payment_id'] ?? null;

        if (!$paymentId) {
            echo "Thiếu payment_id";
            return;
        }

        $this->paymentService->confirmPayment($paymentId);
        include __DIR__ . '/../views/payment/result.php';
    }
}

File: OpticalStore.WebApplication/Controllers/PaymentController.cs
using Microsoft.AspNetCore.Mvc;
using OpticalStore.WebApplication.Models.ViewModels;
using OpticalStore.WebApplication.Services;

namespace OpticalStore.WebApplication.Controllers
{
    public class PaymentController : Controller
    {
        private readonly IPaymentService _paymentService;

        public PaymentController(IPaymentService paymentService)
        {
            _paymentService = paymentService;
        }

        // GET: /Payment/Confirm
        public IActionResult Confirm()
        {
            var model = new PaymentViewModel
            {
                OrderId = 1,
                CustomerName = HttpContext.Session.GetString("UserName") ?? "",
                PaymentMethod = "COD",
                TotalAmount = 1200000
            };

            return View(model);
        }

        // POST: /Payment/Confirm
        [HttpPost]
        public IActionResult Confirm(PaymentViewModel model)
        {
            if (!ModelState.IsValid)
            {
                return View(model);
            }

            var result = _paymentService.ProcessPayment(model);

            if (!result)
            {
                ViewBag.Error = "Thanh toán thất bại. Vui lòng thử lại.";
                return View(model);
            }

            TempData["Success"] = "Thanh toán thành công!";
            return RedirectToAction("Success");
        }

        // GET: /Payment/Success
        public IActionResult Success()
        {
            return View();
        }
    }
}
